cwl/src/main/scala/cwl/CommandLineTool.scala

```diff
@@ -1,66 +1,191 @@
-                                    name: String,
-                                    `type`: MyriadInputType,
-                                    doc: Option[String],
-                                    inputBinding: Option[CommandLineBinding],
-                                    label: Option[String])
-
-case class CommandInputEnumSchema(
-                                   symbols: Array[String],
-                                   `type`: W.`"enum"`.T,
+import common.Checked
+import common.validation.Checked._
+import shapeless.syntax.singleton._
+import shapeless.{:+:, CNil, Witness, _}
+import CommandLineTool._
+import CwlType.CwlType
+import CwlVersion._
+import eu.timepit.refined.W
+import wom.callable.Callable.{OutputDefinition, RequiredInputDefinition}
+import wom.callable.{Callable, TaskDefinition}
+import wom.executable.Executable
+import wom.expression.WomExpression
+import wom.types.WomFileType
+import wom.{CommandPart, RuntimeAttributes}
+
+/**
+  * @param `class` This _should_ always be "CommandLineTool," however the spec does not -er- specify this.
+  */
+case class CommandLineTool private(
+                                   inputs: Array[CommandInputParameter],
+                                   outputs: Array[CommandOutputParameter],
+                                   `class`: Witness.`"CommandLineTool"`.T,
+                                   id: String,
+                                   requirements: Option[Array[Requirement]],
+
+                                   //TODO: Fix this when CwlAny parses correctly
+                                   //hints: Option[Array[CwlAny]] = None,
+                                   hints: Option[Array[Map[String, String]]],
+
                                    label: Option[String],
-                                   inputBinding: Option[CommandLineBinding])
+                                   doc: Option[String],
+                                   cwlVersion: Option[CwlVersion],
+                                   baseCommand: Option[BaseCommand],
+                                   arguments: Option[Array[CommandLineTool.Argument]],
+                                   stdin: Option[StringOrExpression],
+                                   stderr: Option[StringOrExpression],
+                                   stdout: Option[StringOrExpression],
+                                   successCodes: Option[Array[Int]],
+                                   temporaryFailCodes: Option[Array[Int]],
+                                   permanentFailCodes: Option[Array[Int]]) {
+
+  def womExecutable(inputFile: Option[String] = None): Checked[Executable] =
+    CwlExecutableValidation.builWomExecutable(taskDefinition.validNelCheck, inputFile)
+
+  def taskDefinition: TaskDefinition = {
+
+    val id = this.id
+
+    val commandTemplate: Seq[CommandPart] = baseCommand.toSeq.flatMap(_.fold(BaseCommandToCommandParts)) ++
+      arguments.toSeq.flatMap(_.map(_.fold(ArgumentToCommandPart)))
```

Comment:
is this forcompable?

Reply:
Perhaps. But this is just a small rename so I'll leave that for another ticket